pkg/network/capability/capability.go

```diff
@@ -0,0 +1,72 @@
+package capability
+
+import (
+	"errors"
+
+	"github.com/nspcc-dev/neo-go/pkg/io"
+)
+
+// MaxCapabilities is the maximum number of capabilities per payload
+const MaxCapabilities int = 32
+
+// Capability describes network service available for node
+type Capability struct {
+	Type Type
+	Data io.Serializable
+}
+
+// DecodeBinary implements Serializable interface.
+func (c *Capability) DecodeBinary(br *io.BinReader) {
+	c.Type = Type(br.ReadB())
+	switch c.Type {
+	case FullNode:
+		c.Data = &Node{}
+		c.Data.(*Node).DecodeBinary(br)
```

Comment:
No need to type cast here, because `c.Data` is `io.Serializable`.
It can also be called after switch which lead to less code if there will be more variants.